pkg/ccl/backupccl/restore_span_covering_test.go

```diff
@@ -942,13 +971,65 @@ func TestRestoreEntryCover(t *testing.T) {
 									backups[numBackups-1].Spans,
 									backups,
 									layerToIterFactory,
+									nil,
 									target<<20,
 									introducedSpanFrontier,
 									[]jobspb.RestoreProgress_FrontierEntry{},
 									simpleImportSpans)
 								require.NoError(t, err)
 								require.NoError(t, checkRestoreCovering(ctx, backups, backups[numBackups-1].Spans,
 									cover, target != noSpanTargetSize, execCfg.DistSQLSrv.ExternalStorage))
+
+								// Check that the correct import spans are created if the job is
+								// resumed after the completion of some random entries in the cover.
+								if len(cover) > 0 {
+									for n := 1; n <= 5; n++ {
+										var completedSpans []roachpb.Span
+										var highWater []byte
+										if simpleImportSpans {
+											idx := r.Intn(len(cover))
+											completedSpans = append(completedSpans, roachpb.Span{
+												Key:    cover[0].Span.Key,
+												EndKey: cover[idx].Span.EndKey,
+											})
```

Comment:
ack! This made me realize there's a bad bug if `bulkio.restore.use_simple_import_spans` is true on 23.1, where we currently only use the hwm to checkpoint things. But the hwm version of checkpointing is _version_ gated not _feature_ gated. This implies that if use SimpleImportSpans, we don't get any checkpointing goodness.

I'll send out a patch.

Reply:
We should merge [this](https://github.com/cockroachdb/cockroach/pull/99373) before this pr.